include/slang/driver/Driver.h

```diff
@@ -134,6 +135,9 @@ class SLANG_EXPORT Driver {
         /// A set of options controlling translate-off comment directives.
         std::vector<std::string> translateOffOptions;
 
+        /// A map of file patterns to its keyword versions
+        std::map<std::string, parsing::KeywordVersion> keywordMapping;
```

Comment:
I think this actually needs to be a std::vector<std::pair<>> everywhere. You want to maintain the order that the mappings are provided so that there is determinism in which ones get applied if more than one pattern could match a file.